ConcurrentModificationException with --repos from the cli while running info tools

**Describe the bug**
When you use `--repos mavencentral` in the command line, you get a `ConcurrentModificationException` when you run `info tools`

> **Note**
> Our use case is not use `mavencentral` but our company mirror of maven central. But for the reproducer I thought it is easier to use `mavencentral`, because everybody has it.

This is similar to https://github.com/jbangdev/jbang/issues/1377 but the command is not the same `info tools` instead of `run`.

**To Reproduce**
Steps to reproduce the behavior:

Create a small jbang script `log.java`:

```java
///usr/bin/env jbang "$0" "$@" ; exit $?

//DEPS log4j:log4j:1.2.17

import static java.lang.System.out;

import org.apache.log4j.Logger;
import org.apache.log4j.BasicConfigurator;

import java.util.Arrays;

class log {

    static final Logger logger = Logger.getLogger(log.class);

    public static void main(String[] args) {
        BasicConfigurator.configure();
        logger.info("Welcome to jbang");

        Arrays.asList(args).forEach(arg -> logger.warn("arg: " + arg));
        logger.info("Hello from Java!");
    }
}
```

Run it with:
```
 jbang --verbose info tools --fresh --repos mavencentral log.java
```

Usage of `--fresh` make the error appear all the time.
Without `--fresh` it depends (and I could not yet figure out when it isn't working)

jbang is failing with:
```
[jbang] [ERROR] public class java.util.ConcurrentModificationException
```
(see the verbose output at the bottom)

**Expected behavior**
It should work.

**JBang version**

* `0.91.0` 🟢 --> works
* `0.92.0` 🔴 --> first version where this is broken
* `0.92.1` 🔴 --> broken
* `0.92.2` 🔴 --> broken
* `0.93.0` 🔴 --> broken
* `0.94.0` 🔴 --> broken
* `0.95.0` 🔴 --> broken
* `0.96.0` 🔴 --> still broken with `info tools`, 🟢 `run` is working.

**Additional context**
Verbose output:
```
[jbang] jbang version 0.96.0
[jbang] System Java version detected as 8
[jbang] System Java version matches requested version 8
[jbang] Repositories: mavencentral=https://repo1.maven.org/maven2/, mavencentral=https://repo1.maven.org/maven2/
[jbang] Resolving dependencies...
[jbang] [ERROR] public class java.util.ConcurrentModificationException
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)
	at java.util.ArrayList$Itr.next(ArrayList.java:859)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.addRemoteRepo(MavenWorkingSessionImpl.java:315)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:71)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:36)
	at dev.jbang.dependencies.MavenRepo.apply(MavenRepo.java:40)
	at dev.jbang.dependencies.DependencyUtil.lambda$resolveDependenciesViaAether$2(DependencyUtil.java:161)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:161)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:131)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:77)
	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:68)
	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:318)
	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.<init>(Info.java:92)
	at dev.jbang.cli.BaseInfoCommand.getInfo(Info.java:157)
	at dev.jbang.cli.Tools.doCall(Info.java:181)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:136)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:20)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
[jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

I think the main issue is that the same repo is added twice in the `[jbang] Repositories` line.

This is similar to this issue https://github.com/jbangdev/jbang/issues/1377

--- a/build.gradle
+++ b/build.gradle
@@ -48,8 +48,8 @@ dependencies {
 
 	testCompile "com.github.tomakehurst:wiremock-jre8:2.32.0"
 
-	testImplementation "org.junit.jupiter:junit-jupiter:5.7.1"
-	testRuntimeOnly('org.junit.jupiter:junit-jupiter-engine:5.7.1')
+	testImplementation "org.junit.jupiter:junit-jupiter:5.9.0"
+	testRuntimeOnly('org.junit.jupiter:junit-jupiter-engine:5.9.0')
 	testImplementation "com.github.stefanbirkner:system-rules:1.17.2"
 	testImplementation(
 			'org.hamcrest:hamcrest-library:2.2'

--- a/src/main/java/dev/jbang/dependencies/DependencyResolver.java
+++ b/src/main/java/dev/jbang/dependencies/DependencyResolver.java
@@ -19,12 +19,24 @@ public class DependencyResolver {
 	}
 
 	public DependencyResolver addRepository(MavenRepo repository) {
-		repositories.add(repository);
+		MavenRepo repo = repositories	.stream()
+										.filter(r -> r.getId().equals(repository.getId()))
+										.findFirst()
+										.orElse(null);
+		if (repo != null && !repo.getUrl().equals(repository.getUrl())) {
+			throw new IllegalArgumentException("Repository with duplicate id and different url: "
+					+ repository + " vs " + repo);
+		}
+		if (repo == null) {
+			repositories.add(repository);
+		}
 		return this;
 	}
 
 	public DependencyResolver addRepositories(List<MavenRepo> repositories) {
-		this.repositories.addAll(repositories);
+		for (MavenRepo repo : repositories) {
+			addRepository(repo);
+		}
 		return this;
 	}
 

--- a/src/main/java/dev/jbang/dependencies/DependencyUtil.java
+++ b/src/main/java/dev/jbang/dependencies/DependencyUtil.java
@@ -343,6 +343,7 @@ public class DependencyUtil {
 
 		if (split.length == 1) {
 			reporef = split[0];
+			repoid = reporef.toLowerCase();
 		} else if (split.length == 2) {
 			repoid = split[0];
 			reporef = split[1];
@@ -352,7 +353,7 @@ public class DependencyUtil {
 
 		String repo = aliasToRepos.get(reporef.toLowerCase());
 		if (repo != null) {
-			return new MavenRepo(Optional.ofNullable(repoid).orElse(reporef.toLowerCase()), repo);
+			return new MavenRepo(repoid, repo);
 		} else {
 			return new MavenRepo(repoid, reporef);
 		}

--- a/src/main/java/dev/jbang/source/RunContext.java
+++ b/src/main/java/dev/jbang/source/RunContext.java
@@ -304,10 +304,9 @@ public class RunContext {
 
 	/**
 	 * Return resolved classpath lazily. resolution will only happen once, any
-	 * consecutive calls return the same classpath.
-	 *
-	 * Properties available will be used for property replacement.
-	 **/
+	 * consecutive calls return the same classpath. Properties available will be
+	 * used for property replacement.
+	 */
 	public ModularClassPath resolveClassPath(Code code) {
 		if (mcp == null) {
 			DependencyResolver resolver = new DependencyResolver();

--- a/src/test/java/dev/jbang/BaseTest.java
+++ b/src/test/java/dev/jbang/BaseTest.java
@@ -21,7 +21,6 @@ import org.junit.jupiter.api.io.TempDir;
 
 import dev.jbang.cli.BaseCommand;
 import dev.jbang.cli.JBang;
-import dev.jbang.cli.TestRun;
 import dev.jbang.util.Util;
 
 import picocli.CommandLine;
@@ -47,7 +46,7 @@ public abstract class BaseTest {
 
 	@BeforeAll
 	static void init() throws URISyntaxException, IOException {
-		URL examplesUrl = TestRun.class.getClassLoader().getResource(EXAMPLES_FOLDER);
+		URL examplesUrl = BaseTest.class.getClassLoader().getResource(EXAMPLES_FOLDER);
 		if (examplesUrl == null) {
 			examplesTestFolder = Paths.get(EXAMPLES_FOLDER).toAbsolutePath();
 		} else {

--- a/src/test/java/dev/jbang/source/TestRunContext.java
+++ b/src/test/java/dev/jbang/source/TestRunContext.java
@@ -1,0 +1,47 @@
+package dev.jbang.source;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import dev.jbang.BaseTest;
+import dev.jbang.cli.ExitException;
+
+public class TestRunContext extends BaseTest {
+
+	@Test
+	void testDuplicateAnonRepos() {
+		RunContext ctx = new RunContext();
+		ctx.setAdditionalRepositories(Arrays.asList("foo=http://foo", "foo=http://foo"));
+		Path src = examplesTestFolder.resolve("quote.java");
+		Code code = ctx.forFile(src);
+		assertThrows(ExitException.class, () -> {
+			ctx.resolveClassPath(code);
+		});
+	}
+
+	@Test
+	void testDuplicateNamedRepos() {
+		RunContext ctx = new RunContext();
+		ctx.setAdditionalRepositories(Arrays.asList("foo=http://foo", "foo=http://foo"));
+		Path src = examplesTestFolder.resolve("quote.java");
+		Code code = ctx.forFile(src);
+		assertThrows(ExitException.class, () -> {
+			ctx.resolveClassPath(code);
+		});
+	}
+
+	@Test
+	void testReposSameIdDifferentUrl() {
+		RunContext ctx = new RunContext();
+		ctx.setAdditionalRepositories(Arrays.asList("foo=http://foo", "foo=http://bar"));
+		Path src = examplesTestFolder.resolve("quote.java");
+		Code code = ctx.forFile(src);
+		assertThrows(IllegalArgumentException.class, () -> {
+			ctx.resolveClassPath(code);
+		});
+	}
+}

--- a/src/test/resources/junit-platform.properties
+++ b/src/test/resources/junit-platform.properties
@@ -1,0 +1,1 @@
+junit.jupiter.testclass.order.default=org.junit.jupiter.api.ClassOrderer$ClassName

